### PR TITLE
More Pyrotechnics and Science Improvements

### PR DESCRIPTION
**Pull request by AtnerNT**
BTS changes: BYD no longer uses lists of reactions and specific effects to determine bounties. Instead the possible effects of a reaction are cached and passed to the BYD before being activated. This means future developers won't have to tinker with the BYD every time a new smoke, foam or explosive effect is added. It'll now all be handled programmatically.

Progress towards #6156

### Changelog:

CL: [New] Added explosion effects to Chlorine Triflouride, Welding Fuel, Nitro-glycerine and Meth (With minimum temps)
CL: [New] Added flashing effect to flash powder reaction 
CL: [New] Added deafening effect to sonic powder reaction 
CL: [New] Added hearing protection to ear muffs
CL: [New] Added tooltips to the techweb, showing what items a node will unlock by mousing over them
CL: [Balance] Buffed RP gained from artifacts and high energy lab
CL: [Balance] Increased range of chem explosions
CL: [Balance] 'Bluespace Technology' is now a chemistry node
CL: [Balance] Cyborg body parts and the AI brains are now split into two nodes each at a lower cost than the original
CL: [Balance] Mining technology node is now cheaper but lacks the hardsuit design. A new 'Advanced Protokinetics' node is available that unlocks the crusher, machete, spear and claw for production.
CL: [Balance] Anomaly Science node has been removed.
CL: [Balance]  Intellicard now must be unlocked on the techweb before production. Located on the new 'Robotic Intelligence' node
CL: [Improvement] Added various new electronic boards to the techweb under the 'Engineering Electronics' node
CL: [Improvement] Added laser musket and flintlock to the techweb under the 'Civilian Weaponry' node
CL: [Improvement] Added chat notification for bounty completion
CL: [Improvement] Added new explosive bounties
CL: [Improvement] Smaller chem explosions will now be quieter
CL: [Improvement] Each bounty will now be unique, no duplicate bounties.
CL: [Fix] Fixed artifact sample quanities not varying correctly

### CODE REVIEW

**Comment by AtnerNT** [comment]
Singleton test is passing locally, if it fails again I dont know what the issue is

**Comment by Bod9001** [comment]
for me
ScriptableObject (ChemistryReagentsSO) has invalid serializable Unity object fields: 
	Field: allChemistryReactions,at Index 485  has a missing reference.

**Comment by AtnerNT** [comment]
Should now be ready for merge

File: UnityProject/Assets/Scripts/Systems/Chemistry/ReactionSet.cs [hunk 1]
```diff
@@ -103,5 +102,27 @@ public virtual bool Apply(MonoBehaviour sender, ReagentMix reagentMix, List<Reac
 
 			return changed;
 		}
+
+		public static List<CachedEffect> ApplyWithoutEffects(MonoBehaviour sender, ReagentMix reagentMix, HashSet<Reaction> possibleReactions)
+		{
+			bool changing;
+			List<CachedEffect> cachedEffects = new List<CachedEffect>();
+			do
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Systems/Chemistry/ReactionSet.cs:110]: this syntax is weird, but I'll allow it.

**Comment by MaxIsJoe** [comment]
Don't space out changes, it causes the changelog thingy to fail.
fixed it for you.

**Review by MaxIsJoe — APPROVED** [review]

**Pull request merged**